src/main/scala/com/fulcrumgenomics/bam/TrimPrimers.scala

```diff
@@ -142,12 +160,16 @@ class TrimPrimers
     val trimProgress = ProgressLogger(this.logger, "Trimmed")
     while (iterator.hasNext) {
       val reads = nextTemplate(iterator)
-      trimReadsForTemplate(detector, reads)
-      reads.foreach(write)
+      val skip = trimReadsForTemplate(detector, reads)
+      (write_primerless_opt, skip) match {
+        case (Some(write_primerless), true) => reads.foreach(write_primerless)
+        case _ => reads.foreach(write)
+      }
       reads.foreach(trimProgress.record)
     }
 
     // If we had a reference and re-sorted above, reset the NM/UQ/MD tags as we push to the final output
+    //TODO How to implement resorting for primerless reads?
```

Comment:
As I am new to Scala I was not sure how exactly sorting works which is why primerless reads currently remain unsorted when written to the output file.